### PR TITLE
ceph-disk: activate/destroy/list --cluster option

### PR DESCRIPTION
**Pull request by dachary**
This option is required to get the proper mount options. The cluster
will possibly be overriden after the partition is mounted, if the fsid
is different from the provided option.

### CODE REVIEW

**Comment by dachary** [comment]
added tests for the ceph-disk suite

File: src/ceph-disk/ceph_disk/main.py [hunk 1]
```diff
@@ -3494,7 +3495,7 @@ def mount_activate(
 
     # TODO always using mount options from cluster=ceph for
     # now; see http://tracker.newdream.net/issues/3253
-    mount_options = get_mount_options(cluster='ceph', fs_type=fstype)
+    mount_options = get_mount_options(cluster=cluster, fs_type=fstype)
```
Comment (tchaikov) [inline src/ceph-disk/ceph_disk/main.py]: i guess we can remove the "TODO" comment above with this change =)

File: src/ceph-disk/ceph_disk/main.py [hunk 2]
```diff
@@ -5389,6 +5399,12 @@ def make_activate_space_parser(name, subparsers):
         metavar='DEV',
         help='path to %s block device' % name,
     )
+    activate_parser.add_argument(
```
Comment (tchaikov) [inline src/ceph-disk/ceph_disk/main.py]: s/activate_space_parser/activate_parser/

File: src/ceph-disk/ceph_disk/main.py [hunk 3]
```diff
@@ -5436,6 +5452,12 @@ def make_activate_all_parser(subparsers):
         are not activated.
         """)),
         help='Activate all tagged OSD partitions')
+    activate_parser.add_argument(
```
Comment (tchaikov) [inline src/ceph-disk/ceph_disk/main.py]: s/activate_all_parser/activate_parser/

File: src/ceph-disk/ceph_disk/main.py [hunk 4]
```diff
@@ -5465,6 +5487,12 @@ def make_list_parser(subparsers):
         associated Ceph information, if any.
         """)),
         help='List disks, partitions, and Ceph OSDs')
+    activate_parser.add_argument(
```
Comment (tchaikov) [inline src/ceph-disk/ceph_disk/main.py]: s/activate_parser/list_parser/

File: src/ceph-disk/ceph_disk/main.py (hunk 1 above)
Comment (tchaikov) [reply src/ceph-disk/ceph_disk/main.py]: also could you add "Fixes: http://tracker.ceph.com/issues/3253" in the commit message

Comment (dachary) [reply src/ceph-disk/ceph_disk/main.py]: done

File: src/ceph-disk/ceph_disk/main.py (hunk 4 above)
Comment (dachary) [reply src/ceph-disk/ceph_disk/main.py]: done

File: src/ceph-disk/ceph_disk/main.py (hunk 3 above)
Comment (dachary) [reply src/ceph-disk/ceph_disk/main.py]: done

File: src/ceph-disk/ceph_disk/main.py (hunk 2 above)
Comment (dachary) [reply src/ceph-disk/ceph_disk/main.py]: fixed the name

**Comment by dachary** [comment]
@tchaikov thanks for your review, I'll work on the integration tests now

File: src/ceph-disk/ceph_disk/main.py (hunk 3 above)
Comment (tchaikov) [reply src/ceph-disk/ceph_disk/main.py]: @dachary sorry, should be activate_all_parser

Comment (tchaikov) [reply src/ceph-disk/ceph_disk/main.py]: but it's still `activate_parser`. did you forget to push the latest change?

Comment (dachary) [reply src/ceph-disk/ceph_disk/main.py]: yes...

**Comment by dachary** [comment]
logs expired, repushing to get the error

File: src/ceph-disk/ceph_disk/main.py [hunk 5]
```diff
@@ -4540,7 +4542,7 @@ def list_dev(dev, uuid_map, space_map):
     return info
 
 
-def list_devices():
+def list_devices(cluster):
```
Comment (tchaikov) [inline src/ceph-disk/ceph_disk/main.py]: ahh, right. we should update the mock function of `list_devices_return()` as well

**Comment by tchaikov** [comment]
closing in favor of #18638

**Pull request closed without merging**